src/meta/src/rpc/service/notification_service.rs

```diff
@@ -88,16 +92,24 @@ where
                 hummock_version: None,
             },
 
-            WorkerType::Compactor => MetaSnapshot {
-                table,
-                ..Default::default()
-            },
-
-            WorkerType::ComputeNode => MetaSnapshot {
-                table,
-                hummock_version,
-                ..Default::default()
-            },
+            WorkerType::Compactor => {
+                let processing_table = self.stream_manager.processing_table();
```

Comment:
There might be a gap between reading processing table and notifying computer/compactor node.

Reply:
That sounds deadly, how does `Frontend` handle this? 

We have no guarantees of atomicity and lack of re-push, shall we need to actively re-subscription once we found the excetion ?

Reply:
a simple idea, when notify, we first check if some node under re-subscirbe (which has not insert address to sender collection), we temporately store the notification (aovid blocking all the whole process).

After re-subscibe, we re-notify the notification, and we need to check version before handling at observer to avoid updates from being overwritten


it looks like

``` rust

fn notify () {
// 1. notify
self.send();

// 2. check if some node under re-subscirbe

if under_re_subscirbe.load(SeqCst) {
  sub_notifaction.push(notification, version)
}

}

fn subscribe() {
let meta_snapshot = self.build_snapshot_by_type(worker_type).await?;

        tx.send(Ok(SubscribeResponse {
            status: None,
            operation: Operation::Snapshot as i32,
            info: Some(Info::Snapshot(meta_snapshot)),
            version: self.env.notification_manager().current_version().await,
        }))
        .unwrap();

        self.env
            .notification_manager()
            .insert_sender(worker_type, WorkerKey(host_address), tx)
            .await;

       // after insert_sender, the notification are back to normal, so we turn under_re_subscirbe to false
       under_re_subscirbe.store(false)
       // handle sub notification
      
      for n in sub_notification {
            self.notify(notification, version)
     }
}

```